velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -632,14 +632,20 @@ void configureRowReaderOptions(
 namespace {
 
 bool applyPartitionFilter(
+    SplitReader::TableFormat tableFormat,
     const TypePtr& type,
     const std::string& partitionValue,
     common::Filter* filter) {
   if (type->isDate()) {
-    const auto result = util::fromDateString(
-        StringView(partitionValue), util::ParseMode::kPrestoCast);
-    VELOX_CHECK(!result.hasError());
-    return applyFilter(*filter, result.value());
+    int32_t result = 0;
+    // Iceberg partition values are already in daysSinceEpoch, no need to
+    // convert.
+    if (tableFormat == SplitReader::TableFormat::kIceberg) {
+      result = std::stoi(partitionValue);
```

Comment:
The integer types below use `folly::to<int64_t>(partitionValue)`. We could follow them here.